addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -305,11 +305,24 @@ def _export_invoice_vals(self, invoice):
             'due_date': None,
         })
 
+        vals['vals']['accounting_supplier_party_vals']['party_vals'].update(
+            self._l10n_sa_get_supplier_party_vals(invoice))
         vals['vals']['legal_monetary_total_vals'].update(self._l10n_sa_get_monetary_vals(invoice, vals))
         self._l10n_sa_postprocess_line_vals(vals)
 
         return vals
 
+    def _l10n_sa_get_supplier_party_vals(self, invoice):
+        if invoice.journal_id.l10n_sa_branch_id:
+            supplier = invoice.company_id.partner_id.commercial_partner_id
+            return {
+                'party_identification_vals': [{
+                    'id_attrs': {'schemeID': supplier.l10n_sa_additional_identification_scheme},
+                    'id': invoice.journal_id.l10n_sa_branch_id,
+                }]
```

Comment:
It is a bit weird that it being retrieved from 2 different places at once.  One from the company where we risk forgetting to set it and one on the journal.  Is the branch on the journal formatted according to the identification scheme or not?  

The simplest thing might be to  do an extra module l10n_sa_edi_branch.  (to be merged in master and if there are other fields that might be in handy...)